src/Entity/SectionAssociation.php

```diff
@@ -0,0 +1,117 @@
+ *   internal = TRUE,
+ *   handlers = {
+ *     "access" = "Drupal\workbench_access\SectionAssociationAccessControlHandler",
+ *     "storage" = "Drupal\workbench_access\SectionAssociationStorage",
+ *     "views_data" = "\Drupal\workbench_access\ViewsData"
+ *   },
+ *   admin_permission = "assign workbench access",
+ *   base_table = "section_association",
+ *   data_table = "section_association_field_data",
+ *   revision_table = "section_association_revision",
+ *   revision_data_table = "section_association_field_revision_data",
+ *   translatable = FALSE,
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *     "revision" = "vid",
+ *     "bundle" = "access_scheme",
+ *   }
+ * )
+ *
+ * @internal
+ *   This entity is marked internal because it should not be used directly.
+ */
+class SectionAssociation extends ContentEntityBase implements SectionAssociationInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
+    $fields = parent::baseFieldDefinitions($entity_type);
+
+    // Assigned users.
+    $fields['user_id'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Users'))
+      ->setDescription(t('The Name of the associated user.'))
+      ->setSetting('target_type', 'user')
+      ->setSetting('handler', 'default')
+      ->setCardinality(FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED)
+      ->setRevisionable(TRUE)
+      ->setTranslatable(FALSE)
+      ->setDisplayOptions('view', [
+        'label' => 'above',
+        'type' => 'author',
+        'weight' => -2,
+      ])
+      ->setDisplayOptions('form', [
+        'type' => 'entity_reference_autocomplete',
+        'settings' => [
+          'match_operator' => 'CONTAINS',
+          'size' => 60,
+          'placeholder' => '',
+        ],
+        'weight' => -2,
+      ])
+      ->setDisplayConfigurable('form', TRUE)
+      ->setDisplayConfigurable('view', TRUE);
+
+    // Assigned roles.
+    $fields['role_id'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Role'))
+      ->setDescription(t('The roles associated with this section.'))
+      ->setSetting('target_type', 'user_role')
+      ->setSetting('handler', 'default')
+      ->setCardinality(FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED)
+      ->setRevisionable(TRUE)
+      ->setTranslatable(FALSE)
+      ->setDisplayOptions('view', [
+        'label' => 'above',
+        'type' => 'string',
+        'weight' => -2,
+      ])
+      ->setDisplayOptions('form', [
+        'type' => 'entity_reference_autocomplete',
+        'settings' => [
+          'match_operator' => 'CONTAINS',
+          'size' => 60,
+          'placeholder' => '',
+        ],
+        'weight' => -3,
+      ])
+      ->setDisplayConfigurable('form', TRUE)
+      ->setDisplayConfigurable('view', TRUE);
+    $fields['section_id'] = BaseFieldDefinition::create('string')
```

Comment:
We can convert this to ER field. Here is the interdiff https://www.drupal.org/files/issues/interdiff-ea5d49.txt

Reply:
@jibran This is a great review. I'd prefer to commit this as-is here on GitHub and address each of these as follow-up branches / PRs for easier review.

Then we push to drupal.org when all the changes have been made.

We also need some help with the Views integration, since your shorthand comment about DefaultTableView doesn't mean anything to me yet.